pkg/config/sync.go

```diff
@@ -55,6 +61,53 @@ func SetConfigForTesting(config *controller.OperatorConfiguration) {
 	updatePublicConfig()
 }
 
+func ApplyExternalDWOCConfig(workspace *dw.DevWorkspace, client crclient.Client) (err error) {
+
+	if !workspace.Spec.Template.Attributes.Exists(constants.ExternalDevWorkspaceConfiguration) {
+		return nil
+	}
+
+	ExternalDWOCMeta := v1alpha1.ExternalConfig{}
+
+	err = workspace.Spec.Template.Attributes.GetInto(constants.ExternalDevWorkspaceConfiguration, &ExternalDWOCMeta)
+	if err != nil {
+		return fmt.Errorf("failed to read attribute %s in DevWorkspace attributes: %w", constants.ExternalDevWorkspaceConfiguration, err)
+	}
+
+	if ExternalDWOCMeta.Name == "" {
+		return fmt.Errorf("'name' must be set for attribute %s in DevWorkspace attributes", constants.ExternalDevWorkspaceConfiguration)
+	}
+
+	if ExternalDWOCMeta.Namespace == "" {
+		return fmt.Errorf("'namespace' must be set for attribute %s in DevWorkspace attributes", constants.ExternalDevWorkspaceConfiguration)
+	}
+
+	externalDWOC := &controller.DevWorkspaceOperatorConfig{}
+	namespacedName := types.NamespacedName{
+		Name:      ExternalDWOCMeta.Name,
+		Namespace: ExternalDWOCMeta.Namespace,
+	}
+
+	err = client.Get(context.TODO(), namespacedName, externalDWOC)
+	if err != nil {
+		return fmt.Errorf("could not fetch external DWOC with name '%s' in namespace '%s': %w", ExternalDWOCMeta.Name, ExternalDWOCMeta.Namespace, err)
+	}
+
+	// TODO: It might be better to just always merge rather than suffering performance hit in configsAlreadyMerged() (or maintaing configsAlreadyMerged if we don't use reflect.DeepEqual)
+	if !configsAlreadyMerged(externalDWOC.Config, internalConfig) {
```

Comment:
I would guess that given the current design, `configsAlreadyMerged` will almost always return false if there are multiple workspaces.